src/vs/workbench/contrib/notebook/common/model/notebookTextModel.ts

```diff
@@ -207,6 +207,16 @@ export class NotebookTextModel extends Disposable implements INotebookTextModel
 		return this._cells;
 	}
 
+	*cellsForView(): Iterable<NotebookCellTextModel> {
+		if (this.viewType === 'repl') {
```

Comment:
This is only used at the view/viewmodel layer, I'll suggest we move this to `NotebookViewModel`.